[Jonus Ho Zhi Wen] iP

jonushzw opened the pull request:
# Wolfie - Task Management Chatbot 🚀
> "Wolfie is the personal assistant you've been waiting for." - Tech Innovators

Wolfie is a chatbot designed to help you manage your tasks efficiently. This PR introduces new features to enhance your productivity.

## Wolfie has the competitive edge over it's competitors
- *Easy To Use*
- *Fast Response*
- *~~Small~~ EXTRA SMALL Size*

## Steps to Test
1. Clone the repository.
2. Navigate to the project directory.
3. Run the application using the command:

   ```bash
   ./gradlew run

## Features Added:
- [ ] 📝**Task Management**
- [ ] 📅 **Event Scheduling**
- [ ] ⏰**Deadline Tracking**

For more details, visit the [Wolfie Documentation](https://github.com/jonushzw/ip).

## Main Method Example

Here's the `main` method:

```java
public class Wolfie {
    public static void main(String[] args) {
        new Wolfie("data/tasks.txt").run();
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jonushzw committed:
Level-0 Commit

jonushzw committed:
Level-1 Commit

jonushzw committed:
Level-2 Commit

jonushzw committed:
Level-3 Commit Mark as Done

jonushzw committed:
Level-4 Commit ToDos, Events, Deadlines

jonushzw committed:
A-TextUiTesting

jonushzw committed:
Level 5 commit Handling Errors

jonushzw committed:
Level 6 commit Delete and added a welcome text art

jonushzw committed:
Commit A-Enums

jonushzw committed:
Commit 1 for Level-7

jonushzw committed:
Commit 2 for Level-7 saved data for tasks

jonushzw committed:
Commit 3 for Level-7 check if saved data for tasks is updated with every program run

jonushzw committed:
Commit for Level-8 Modified Deadline and Event classes to format dates

jonushzw committed:
Commit for Level-8 Modified Wolfie and subclasses on method still not working properly

jonushzw committed:
Commit for Level-8 Fixed on if else statement

jonushzw committed:
Commit for A-MoreOOP extracted classes for more OOP check for remaining exceptions

jonushzw committed:
Commit for A-MoreOOP Updated variable fields

jonushzw committed:
Commit for A-Packages Organised into Packages and updated package source access

jonushzw committed:
Merge branch 'add-gradle-support'

jonushzw committed:
Added checkstyles.xml and changed Ui.java and Wolfie.java to suit styling

jonushzw committed:
changed all the files to suit style convention

jonushzw committed:
Added tests for MarkCommandTest and StorageTest

jonushzw committed:
Added tests for parser and ui

jonushzw committed:
Added tests for all commands

jonushzw committed:
Updated OnCommandTest.java and UiTest.java

jonushzw committed:
Updated build.gradle for the correct directories for JAR file

jonushzw committed:
Added JavaDocs for remaining methods not highlighted by checkstyle

jonushzw committed:
Added Comments and aligned some lines to follow coding standard

jonushzw committed:
Merge branch 'A-CodingStandard'

jonushzw committed:
Added Find as a command and added a test file

jonushzw committed:
Add JavaFX implementation barebones

jonushzw committed:
Add changes to command methods to return String values

jonushzw committed:
Add image of a wolf to chatbot

jonushzw committed:
Add varArgs for showTasksOnDate method

jonushzw committed:
Add javadocs comments to methods

martout2002 commented on src/main/java/wolfie/Wolfie.java:
```diff
@@ -0,0 +1,71 @@
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
```
Good javadocs comment, easily understandable on how to use Wolfie class

martout2002 commented on src/main/java/wolfie/command/AddEventCommand.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws IOException, WolfieException {
```
could have an @param to show what execute takes in, but overall very well done!

martout2002 reviewed (commented):
LGTM practiced OO principles well

achl1012 commented on src/main/java/wolfie/Wolfie.java:
```diff
@@ -0,0 +1,71 @@
+
+    /**
+     * Generates a response for the user's chat message.
+     */
```
Good use of Javadoc comments, code also adheres to coding standards

achl1012 commented on src/main/java/wolfie/command/FindCommand.java:
```diff
@@ -0,0 +1,34 @@
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws IOException, WolfieException {
+        List<Task> matchingTasks = tasks.getTasks().stream()
+                .filter(task -> task.getDescription().contains(keyword))
+                .toList();
```
good use of line breaks here to make the code neater

achl1012 commented on src/main/java/wolfie/command/Command.java:
```diff
@@ -0,0 +1,32 @@
+     * @return The response to the user
+     * @throws IOException     If an I/O error occurs
+     * @throws WolfieException If an error occurs while executing the command
+     */
```
Good use of Javadoc comments here for the execute method! This could be expanded to include similar Javadoc comments for the execute methods in all the subclasses of the abstract Command class

achl1012 reviewed (commented):
LGTM! Code adheres to the coding standards and the Javadoc comments are descriptive and make the code easier to read and understand.

jonushzw committed:
Add assertions to important methods

Assertions were implemented to fields which require field inputs such as FindCommand or MainWindow

Making sure assertions are input help catch errors

As a measure, will add more assertion checks in the future and into all commands

jonushzw committed:
Merge pull request #1 from jonushzw/branch-A-Assertions

Add assertions to important methods

jonushzw committed:
Add code quality improvements

After reading reviews from peers about improvements I should make, I added the appropriate changes.

Added javadocs to command classes and their respective methods.

jonushzw committed:
Merge pull request #2 from jonushzw/A-CodeQuality

Add code quality improvements

jonushzw committed:
Add streams into Wolfie

Implement stream use in some methods that require listing or a filter.

ListCommand and FindCommand use streams to get their respective lists to display in the GUI

jonushzw committed:
Merge pull request #3 from jonushzw/A-Streams

Add streams into Wolfie

jonushzw committed:
Add CI to project

jonushzw committed:
Fix test files for command methods

jonushzw committed:
Add detection for duplicates

Changed equals method to check for matching names.

Each task type has a boolean check to check for a repeat of the same task.

If the task already exists, a message saying it already exists is printed and the task will not be added.

jonushzw committed:
Merge pull request #4 from jonushzw/C-DetectDuplicates

Add detection for duplicates

jonushzw committed:
Add GUI improvements

Tweaked the colours and paddings to be more vibrant for the users

Added circular profile pictures

jonushzw committed:
Add resizing of windows

The window now resizes properly when resized

Text field adjusts according to resize

jonushzw committed:
Add thread sleep for 5 seconds

This is so the program does not end abruptly

jonushzw committed:
Change github CI

jonushzw committed:
Highlight errors when the user types an invalid command

jonushzw committed:
Merge pull request #5 from jonushzw/A-BetterGui

A better Gui

jonushzw committed:
Add error handling for more cases

Environment issues: Add try catch blocks to check for files that do not exist

Deadline and Task subclasses: Add time check

jonushzw committed:
Merge pull request #6 from jonushzw/A-MoreErrorHandling

Add error handling for more cases

jonushzw committed:
Add testing for multiple classes

jonushzw committed:
Add more testing

Add testing for more methods to check if edge cases work e.g empty file

jonushzw committed:
Merge pull request #8 from jonushzw/A-MoreTesting

A More Testing

jonushzw committed:
Add more error handling

jonushzw committed:
Add more error handling

jonushzw committed:
Add Ui picture

jonushzw committed:
Change Ui picture

jonushzw committed:
Update README

Changed README for commit

jonushzw committed:
Merge pull request #9 from jonushzw/A-UserGuide

Update README

jonushzw committed:
Update GUI

Added more personality to the GUI

Changed User Icon from default

jonushzw committed:
Update gradle config file

Add assertions to be included in the build file

Ensured that assertions will be tested for future runs

jonushzw committed:
Update window size

Tested run on windows computer window size was out of bounds

Made the GUI smaller

jonushzw committed:
Update javadoc comments

Ensured that more than half the public methods/classes have comments
